scripts/jenkins/cloud/ansible/deploy-caasp-v4-terraform.yml

```diff
@@ -0,0 +1,565 @@
+      - { dest: "{{ tfvars_path }}", regexp: '^master_size = "', line: 'master_size = "{{ master_size }}"' }
+      - { dest: "{{ tfvars_path }}", regexp: '^worker_size = "', line: 'worker_size = "{{ worker_size }}"' }
+
+  - name: Management Machine | Update terraform.tfvars authorized_keys
+    replace:
+      path: "{{ tfvars_path }}"
+      regexp: '^authorized_keys = \[\n.*\n]$'
+      replace: 'authorized_keys = [\n  "{{ publickey.stdout }}"\n]'
+
+  - name: Management Machine | Update terraform.tfvars repositories
+    replace:
+      path: "{{ tfvars_path }}"
+      regexp: '^repositories = {}$'
+      replace: 'repositories = {  {{ repositories }} }'
+
+  - name: Management Machine | Update terraform.tfvars number of master nodes
+    replace:
+      path: "{{ tfvars_path }}"
+      regexp: '^masters = .*$'
+      replace: 'masters = {{ num_masters | int }}'
+
+  - name: Management Machine | Update terraform.tfvars number of worker nodes
+    replace:
+      path: "{{ tfvars_path }}"
+      regexp: '^workers = .*$'
+      replace: 'workers = {{ num_workers | int }}'
+
+  - name: Management Machine | Update cpi.auto.tfvars to enable cloud provider integration with Openstack
+    replace:
+      path: "{{ cpiautotfvars_path }}"
+      regexp: '^#cpi_enable = true$'
+      replace: 'cpi_enable = true'
+
+  - name: Management Machine | Update cpi.auto.tfvars to set up the ca_file
+    replace:
+      path: "{{ cpiautotfvars_path }}"
+      regexp: '^#ca_file = "/etc/pki/trust/anchors/.*"$'
+      replace: 'ca_file = "/etc/pki/trust/anchors/ca-bundle.pem"'
+
+  #Update network.tf to set up dns servers so the cloud-init on the caasp nodes does not fail
+  - name: Management Machine | Update network.tf to set up dns servers
+    lineinfile:
+      path: "{{ network_tf_path }}"
+      insertafter: 'resource "openstack_networking_subnet_v2" "subnet" {'
+      line: '  dns_nameservers = [ "10.84.2.20", "10.84.2.21"]'
+
+  #Update load-balancer.tf to use haproxy provider in SOC 8
+  #For SOC 9, we use octavia provider by default
+  #For SOC 8, we use haproxy provider since the octavia provider in SOC 8 has slowness that cause terraform to fail
+  - name: Management Machine | Update load-balancer.tf to setup haproxy provider
+    lineinfile:
+      path: "{{ lb_tf_path }}"
+      insertafter: 'resource "openstack_lb_loadbalancer_v2" "lb" {'
+      line: '  loadbalancer_provider = "haproxy"'
+    when: hostvars[cloud_env]['ansible_distribution_release'] == '3'
+
+  - name: Management Machine | Run terraform init
+    shell: |
+      set -e
+      cd {{ caasp_openstack_path }}
+      terraform init | tee -a terraform_init.log
+    register:
+      terraform_init_results
+    failed_when: "not ('Terraform has been successfully initialized!' in terraform_init_results.stdout)"
+
+  - name: Management Machine |  Run terraform plan
+    shell: |
+      set -e
+      eval `ssh-agent -s`
+      ssh-add
```

Comment:
Could the host machine be set up so it starts ssh-agent automatically, so you do not need to invoke it in each terraform call?